### PR TITLE
[camera] Camera fix android audio

### PR DESCRIPTION
**Pull request by mvanbeusekom**
## Description

Extends on the work of @cfchris from pull request #2426 by extracting the logic of creating the `MediaRecord` instance into a `MediaRecordBuilder` class (using the builder pattern).  This allows us to isolate the code and added the requested unit-tests. 

If this pull request is accepted, pull request #2426 will become obsolete.

## Related Issues

Fixes flutter/flutter#38787

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by mvanbeusekom** [comment]
@googlebot I consent.

**Review by tvolkert — APPROVED** [review]
LGTM

**Comment by cfchris** [comment]
@mvanbeusekom I cloned your fork and tested the example app. It works as expected now. Thank you.

File: packages/camera/android/build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 group 'io.flutter.plugins.camera'
 version '1.0-SNAPSHOT'
-def args = ["-Xlint:deprecation","-Xlint:unchecked","-Werror"]
+def args = ["-Xlint:deprecation","-Xlint:unchecked"]
```
Comment (amirh) [inline packages/camera/android/build.gradle:3]: Why do we need to remove `-Werror`? are we hitting warnings we cannot fix?

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+package io.flutter.plugins.camera.media;
```
Comment (amirh) [inline packages/camera/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java:1]: nit: missing license header

**Review by amirh — COMMENTED** [review]
Left 2 nits, otherwise looks good. Thanks!

File: packages/camera/android/build.gradle (hunk 1 above)
Comment (mvanbeusekom) [reply packages/camera/android/build.gradle:3]: I was running into some warning when running the (already existing) `DartMessengerTest` locally. I shouldn't have checked this in.

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java (hunk 2 above)
Comment (mvanbeusekom) [reply packages/camera/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java:4]: Added missing license header

**Review by amirh — APPROVED** [review]
LGTM

**Pull request merged**